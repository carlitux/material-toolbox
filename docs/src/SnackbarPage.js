// @flow
/* eslint-disable */
import * as React from 'react';
import Highlighter from './Highlighter';

import { LayoutCell } from 'material-toolbox/layout-grid';
import { Text } from 'material-toolbox/typography';
import Theme from 'material-toolbox/theme';
import { Button } from 'material-toolbox/button';
import { List, ListItem } from 'material-toolbox/list';
import  Snackbar  from 'material-toolbox/snackbar';
/* eslint-enable */

import ThemedLink from './ThemedLink';
import styles from './styles.scss';

const importComponents = `import Snackbar from 'material-toolbox/snackbar';`;
const cbSignature = '() => void';

export default class SnackbarPage extends React.Component<
  {},
  {
    show: boolean,
    multiline: boolean,
    onBottom: boolean,
    dismiss: boolean,
    aligned: boolean,
  },
> {
  state = {
    show: false,
    multiline: false,
    onBottom: false,
    dismiss: true,
    aligned: false,
  };

  componentDidMount() {
    window.scrollTo(0, 0);
  }

  showSnackbar = () => {
    this.setState({ show: true, aligned: false });
  };

  showSnackbarAligned = () => {
    this.setState({ show: true, aligned: true });
  };

  render() {
    return (
      <LayoutCell align="bottom" span={12}>
        <Text component="h1" textStyle="display1">
          Snackbar
        </Text>

        <Text component="h2" textStyle="headline">
          Snackbar component is a React wrapper of mdc-snackbar component.
        </Text>

        <Text component="h3" textStyle="title">
          Design & API Documentation
        </Text>

        <p>
          <ThemedLink
            to="https://material.io/guidelines/components/snackbars-toasts.html"
            target="_blank">
            Material Design guidelines: Snackbars & toasts
          </ThemedLink>
        </p>

        <p>
          <ThemedLink
            to="https://material.io/components/web/catalog/snackbars/"
            target="_blank">
            Material Components Web Css and Scss customization
          </ThemedLink>
        </p>

        <Text component="h1" textStyle="display1">
          React components
        </Text>

        <Text component="h3" textStyle="title">
          Import
        </Text>

        <Highlighter language="javascript">{importComponents}</Highlighter>

        <Text component="h3" textStyle="title">
          Snackbar
        </Text>

        <table className={styles['table-doc']}>
          <thead>
            <tr>
              <th>Property</th>
              <th>Type</th>
              <th>Default</th>
              <th>Required</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>message</td>
              <td>string</td>
              <td />
              <td>âœ”</td>
              <td>message to be show</td>
            </tr>
            <tr>
              <td>actionText</td>
              <td>string</td>
              <td />
              <td />
              <td>label to put in action button</td>
            </tr>
            <tr>
              <td>align</td>
              <td>boolean</td>
              <td />
              <td />
              <td>If set to true the snackbar is at left</td>
            </tr>
            <tr>
              <td>active</td>
              <td>boolean</td>
              <td />
              <td />
              <td>True to display the if a prop is changed</td>
            </tr>
            <tr>
              <td>multiline</td>
              <td>boolean</td>
              <td />
              <td />
              <td>Set multiline design</td>
            </tr>
            <tr>
              <td>actionOnBottom</td>
              <td>boolean</td>
              <td />
              <td />
              <td>
                If multiline and if set this to true the action is set at
                bottom.
              </td>
            </tr>
            <tr>
              <td>dismiss</td>
              <td>boolean</td>
              <td>true</td>
              <td />
              <td>if set to false snackbar will stay visible</td>
            </tr>
            <tr>
              <td>timeout</td>
              <td>number</td>
              <td>2750</td>
              <td />
              <td>The time that snackbar is visible</td>
            </tr>
            <tr>
              <td>onActionClick</td>
              <td>{cbSignature}</td>
              <td />
              <td />
              <td>If actionText set needs to set this with a callback</td>
            </tr>
            <tr>
              <td>onShow</td>
              <td>{cbSignature}</td>
              <td />
              <td />
              <td>Notify snackbar has been shown.</td>
            </tr>
            <tr>
              <td>onHide</td>
              <td>{cbSignature}</td>
              <td />
              <td />
              <td>Notify snackbar has been hidden.</td>
            </tr>
          </tbody>
        </table>

        <Button buttonVariant="raised" onClick={this.showSnackbar}>
          Show
        </Button>

        <Button buttonVariant="raised" onClick={this.showSnackbarAligned}>
          Show Aligned
        </Button>

        <label htmlFor="multiline">
          <input
            id="multiline"
            value={this.state.multiline}
            type="checkbox"
            onChange={() => {
              this.setState({ multiline: !this.state.multiline });
            }}
          />{' '}
          Multiline
        </label>

        <label htmlFor="bottom">
          <input
            id="bottom"
            value={this.state.onBottom}
            disabled={!this.state.multiline}
            type="checkbox"
            onChange={() => {
              this.setState({ onBottom: !this.state.onBottom });
            }}
          />{' '}
          Action On Bottom
        </label>

        <label htmlFor="dismiss">
          <input
            id="dismiss"
            value={this.state.dismiss}
            defaultChecked={this.state.dismiss}
            type="checkbox"
            onChange={() => {
              this.setState({ dismiss: !this.state.dismiss });
            }}
          />{' '}
          Dismiss on Action
        </label>

        <Snackbar
          message="This a example text"
          actionText="Undo"
          timeout={3000}
          multiline={this.state.multiline}
          actionOnBottom={this.state.onBottom}
          active={this.state.show}
          dismiss={this.state.dismiss}
          align={this.state.aligned}
          onActionClick={() => {
            console.log('clicked');
          }}
        />
      </LayoutCell>
    );
  }
}
