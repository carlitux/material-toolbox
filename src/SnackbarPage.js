// @flow
import * as React from 'react';

import { LayoutCell } from './material-toolbox/layout-grid';
import { Text } from './material-toolbox/typography';

import ThemedLink from './ThemedLink';
import BasicExample from './snackbar-examples/BasicExample';

const cbSignature = '() => void';

type Props = {};

type State = {};

export default class SnackbarPage extends React.Component<Props, State> {
  componentDidMount() {
    window.scrollTo(0, 0);
  }

  render() {
    return (
      <LayoutCell align="bottom" span={12}>
        <Text component="h1" textStyle="display1">
          Snackbar
        </Text>

        <Text component="h2" textStyle="headline">
          Snackbar component is a React wrapper of mdc-snackbar component.
        </Text>

        <Text component="h3" textStyle="title">
          Design & API Documentation
        </Text>

        <p>
          <ThemedLink
            to="https://material.io/guidelines/components/snackbars-toasts.html"
            target="_blank">
            Material Design guidelines: Snackbars & toasts
          </ThemedLink>
        </p>

        <p>
          <ThemedLink
            to="https://material.io/components/web/catalog/snackbars/"
            target="_blank">
            Material Components Web Css and Scss customization
          </ThemedLink>
        </p>

        <Text component="h1" textStyle="display1">
          React Components
        </Text>

        <Text component="h3" textStyle="title">
          Snackbar
        </Text>

        <table className="table-props">
          <thead>
            <tr>
              <th>Property</th>
              <th>Type</th>
              <th>Default</th>
              <th>Required</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>message</td>
              <td>string</td>
              <td />
              <td>âœ”</td>
              <td>message to be show</td>
            </tr>
            <tr>
              <td>actionText</td>
              <td>string</td>
              <td />
              <td />
              <td>label to put in action button</td>
            </tr>
            <tr>
              <td>align</td>
              <td>boolean</td>
              <td />
              <td />
              <td>If set to true the snackbar is at left</td>
            </tr>
            <tr>
              <td>active</td>
              <td>boolean</td>
              <td />
              <td />
              <td>True to display the if a prop is changed</td>
            </tr>
            <tr>
              <td>multiline</td>
              <td>boolean</td>
              <td />
              <td />
              <td>Set multiline design</td>
            </tr>
            <tr>
              <td>actionOnBottom</td>
              <td>boolean</td>
              <td />
              <td />
              <td>
                If multiline and if set this to true the action is set at
                bottom.
              </td>
            </tr>
            <tr>
              <td>dismissesOnAction</td>
              <td>boolean</td>
              <td>true</td>
              <td />
              <td>if set to false snackbar will stay visible</td>
            </tr>
            <tr>
              <td>timeout</td>
              <td>number</td>
              <td>2750</td>
              <td />
              <td>The time that snackbar is visible</td>
            </tr>
            <tr>
              <td>onClick</td>
              <td>{cbSignature}</td>
              <td />
              <td />
              <td>If actionText set needs to set this with a callback</td>
            </tr>
            <tr>
              <td>onShow</td>
              <td>{cbSignature}</td>
              <td />
              <td />
              <td>Notify snackbar has been shown.</td>
            </tr>
            <tr>
              <td>onHide</td>
              <td>{cbSignature}</td>
              <td />
              <td />
              <td>Notify snackbar has been hidden.</td>
            </tr>
          </tbody>
        </table>

        <BasicExample />
      </LayoutCell>
    );
  }
}
