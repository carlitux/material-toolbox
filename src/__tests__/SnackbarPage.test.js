// @flow
import React from 'react';
import { shallow } from 'enzyme';
import SnackbarPage from '../SnackbarPage';

describe('page::SnackbarPage', () => {
  it('Should match snapshoot', () => {
    const scrollTo = jest.fn();
    window.scrollTo = scrollTo;
    const component = shallow(<SnackbarPage />);
    expect(component).toMatchSnapshot();
    expect(scrollTo.mock.calls).toHaveLength(1);
    delete window.scrollTo;
  });
});
