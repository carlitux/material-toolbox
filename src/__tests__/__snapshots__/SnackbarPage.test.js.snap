// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page::SnackbarPage Should match snapshoot 1`] = `
<Cell
  align="bottom"
  span={12}
>
  <Text
    adjustMargin={false}
    component="h1"
    textStyle="display1"
  >
    Snackbar
  </Text>
  <Text
    adjustMargin={false}
    component="h2"
    textStyle="headline"
  >
    Snackbar component is a React wrapper of mdc-snackbar component.
  </Text>
  <Text
    adjustMargin={false}
    component="h3"
    textStyle="title"
  >
    Design & API Documentation
  </Text>
  <p>
    <ThemedLink
      component="a"
      target="_blank"
      to="https://material.io/guidelines/components/snackbars-toasts.html"
    >
      Material Design guidelines: Snackbars & toasts
    </ThemedLink>
  </p>
  <p>
    <ThemedLink
      component="a"
      target="_blank"
      to="https://material.io/components/web/catalog/snackbars/"
    >
      Material Components Web Css and Scss customization
    </ThemedLink>
  </p>
  <Text
    adjustMargin={false}
    component="h1"
    textStyle="display1"
  >
    React Components
  </Text>
  <Text
    adjustMargin={false}
    component="h3"
    textStyle="title"
  >
    Snackbar
  </Text>
  <table
    className="table-props"
  >
    <thead>
      <tr>
        <th>
          Property
        </th>
        <th>
          Type
        </th>
        <th>
          Default
        </th>
        <th>
          Required
        </th>
        <th>
          Description
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          message
        </td>
        <td>
          string
        </td>
        <td />
        <td>
          âœ”
        </td>
        <td>
          message to be show
        </td>
      </tr>
      <tr>
        <td>
          actionText
        </td>
        <td>
          string
        </td>
        <td />
        <td />
        <td>
          label to put in action button
        </td>
      </tr>
      <tr>
        <td>
          align
        </td>
        <td>
          boolean
        </td>
        <td />
        <td />
        <td>
          If set to true the snackbar is at left
        </td>
      </tr>
      <tr>
        <td>
          active
        </td>
        <td>
          boolean
        </td>
        <td />
        <td />
        <td>
          True to display the if a prop is changed
        </td>
      </tr>
      <tr>
        <td>
          multiline
        </td>
        <td>
          boolean
        </td>
        <td />
        <td />
        <td>
          Set multiline design
        </td>
      </tr>
      <tr>
        <td>
          actionOnBottom
        </td>
        <td>
          boolean
        </td>
        <td />
        <td />
        <td>
          If multiline and if set this to true the action is set at bottom.
        </td>
      </tr>
      <tr>
        <td>
          dismissesOnAction
        </td>
        <td>
          boolean
        </td>
        <td>
          true
        </td>
        <td />
        <td>
          if set to false snackbar will stay visible
        </td>
      </tr>
      <tr>
        <td>
          timeout
        </td>
        <td>
          number
        </td>
        <td>
          2750
        </td>
        <td />
        <td>
          The time that snackbar is visible
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          () =&gt; void
        </td>
        <td />
        <td />
        <td>
          If actionText set needs to set this with a callback
        </td>
      </tr>
      <tr>
        <td>
          onShow
        </td>
        <td>
          () =&gt; void
        </td>
        <td />
        <td />
        <td>
          Notify snackbar has been shown.
        </td>
      </tr>
      <tr>
        <td>
          onHide
        </td>
        <td>
          () =&gt; void
        </td>
        <td />
        <td />
        <td>
          Notify snackbar has been hidden.
        </td>
      </tr>
    </tbody>
  </table>
  <BasicExample />
</Cell>
`;
